executor/src/witgen/sequence_iterator.rs

```diff
@@ -167,29 +167,34 @@ where
 }
 
 pub enum ProcessingSequenceIterator {
+    /// The default strategy
     Default(DefaultSequenceIterator),
+    /// The machine has been run successfully before and the sequence is cached.
     Cached(<Vec<SequenceStep> as IntoIterator>::IntoIter),
```

Comment:
Interesting! I somehow was not aware that one can use `IntoIterator` as an "owning" iterator.

Why this and not just `Vec<SequenceStep>`? Because all of them are iterators?

Reply:
`ProcessingSequenceIterator` implements `Iterator`, so it can just forward the call to `next()`. `Vec<_>` does not have next, so that wouldn't work, right?